### PR TITLE
Simplify flow types distribution

### PR DESCRIPTION
**Pull request by TrySound**
Simple referencing to source is in a way cleaner way to distribute flow typed than copying source code over and over again.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
   "main": "lib/index.js",
   "files": [
     "/lib",
-    "/dist"
+    "/dist",
+    "/src"
```
Comment (alexreardon) [inline package.json]: do we need to copy all of the src directory to npm for this?

File: package.json [hunk 2]
```diff
@@ -29,7 +30,7 @@
     "build:clean": "rimraf lib && rimraf dist",
     "build:dist": "rollup -c && rollup -c --min",
     "build:lib": "babel src -d lib",
-    "build:flow": "flow-copy-source --verbose src lib",
+    "build:flow": "echo \"//@flow\n\nexport * from '../src';\" > lib/index.js.flow",
```
Comment (alexreardon) [inline package.json]: super interesting!

@jamiebuilds what do you think?

File: package.json (hunk 1 above)
Comment (TrySound) [reply package.json]: You still need source code with types. Just don't need to copy it to different folder.

Comment (alexreardon) [reply package.json]: Oh I see. Hence the importing from `../src`. Does it seem like an antipattern to you for lib to import from src?

Comment (TrySound) [reply package.json]: This is just a trick. Sources are not imported in actual code. Also for me published source is just a way to observe code from node_modules without opening github.

File: package.json (hunk 2 above)
Comment (lukebatchelor) [reply package.json]: Ha. Thats an interesting one. Nice

Comment (Andarist) [reply package.json]: Leaving this as reference - https://blog.kentcdodds.com/distributing-flow-type-definitions-for-node-and-browser-modules-3952ad38b357

**Comment by alexreardon** [comment]
I will try to get this merged tomorrow

**Comment by alexreardon** [comment]
It looks like there a minor conflict in package.json

**Comment by SimenB** [comment]
I _think_ this is giving me a ton of flow errors from within the source. Using 0.67.1 of flow. Using version 5 of this lib gives no errors

```
Error ------------------------- node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:372:36

Cannot assign `requestAnimationFrame(...)` to `collectFrameId` because `AnimationFrameID` [1] is incompatible with
number [2].

   node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:372:36
                                            v----------------------------
    372|     const collectFrameId: number = requestAnimationFrame(() => {
    373|       const toBePublishedBuffer: UnknownDimensionType[] = toBeCollected.map(
    374|         (descriptor: UnknownDescriptorType): UnknownDimensionType => {
    375|           // is a droppable
    376|           if (descriptor.type) {
    377|             return state.droppables[descriptor.id].callbacks.getDimension();
    378|           }
    379|           // is a draggable
    380|           return state.draggables[descriptor.id].getDimension();
    381|         }
    382|       );
    383|
    384|       // Phase 2: publish all dimensions to the store
    385|       const publishFrameId: number = requestAnimationFrame(() => {
    386|         const toBePublished: ToBePublished = toBePublishedBuffer.reduce(
       :
    399|           toBePublished.droppables,
    400|           toBePublished.draggables,
    401|         );
    402|
    403|         // need to watch the scroll on each droppable
    404|         toBePublished.droppables.forEach((dimension: DroppableDimension) => {
    405|           const entry: DroppableEntry = state.droppables[dimension.descriptor.id];
    406|           entry.callbacks.watchScroll(request.scrollOptions);
    407|         });
    408|
    409|         setFrameId(null);
    410|       });
    411|
    412|       setFrameId(publishFrameId);
    413|     });
             -^

References:
   /private/tmp/flow/flowlib_5c03ba9/dom.js:3265:80
   3265| declare function requestAnimationFrame(callback: (timestamp: number) => void): AnimationFrameID;
                                                                                        ^^^^^^^^^^^^^^^^ [1]
   node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:372:27
    372|     const collectFrameId: number = requestAnimationFrame(() => {
                                   ^^^^^^ [2]


Error ------------------------- node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:385:38

Cannot assign `requestAnimationFrame(...)` to `publishFrameId` because `AnimationFrameID` [1] is incompatible with
number [2].

   node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:385:38
                                              v----------------------------
    385|       const publishFrameId: number = requestAnimationFrame(() => {
    386|         const toBePublished: ToBePublished = toBePublishedBuffer.reduce(
    387|           (previous: ToBePublished, dimension: UnknownDimensionType): ToBePublished => {
    388|             // is a draggable
    389|             if (dimension.placeholder) {
    390|               previous.draggables.push(dimension);
    391|             } else {
    392|               previous.droppables.push(dimension);
    393|             }
    394|             return previous;
    395|           }, { draggables: [], droppables: [] }
    396|         );
    397|
    398|         callbacks.bulkPublish(
    399|           toBePublished.droppables,
    400|           toBePublished.draggables,
    401|         );
    402|
    403|         // need to watch the scroll on each droppable
    404|         toBePublished.droppables.forEach((dimension: DroppableDimension) => {
    405|           const entry: DroppableEntry = state.droppables[dimension.descriptor.id];
    406|           entry.callbacks.watchScroll(request.scrollOptions);
    407|         });
    408|
    409|         setFrameId(null);
    410|       });
               -^

References:
   /private/tmp/flow/flowlib_5c03ba9/dom.js:3265:80
   3265| declare function requestAnimationFrame(callback: (timestamp: number) => void): AnimationFrameID;
                                                                                        ^^^^^^^^^^^^^^^^ [1]
   node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:385:29
    385|       const publishFrameId: number = requestAnimationFrame(() => {
                                     ^^^^^^ [2]


Error ------------------------- node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:425:28

Cannot call `cancelAnimationFrame` with `state.frameId` bound to `requestId` because number [1] is incompatible with
`AnimationFrameID` [2].

   node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:425:28
    425|       cancelAnimationFrame(state.frameId);
                                    ^^^^^^^^^^^^^

References:
   node_modules/react-beautiful-dnd/src/state/dimension-marshal/dimension-marshal.js:36:13
     36|   frameId: ?number,
                     ^^^^^^ [1]
   /private/tmp/flow/flowlib_5c03ba9/dom.js:3266:50
   3266| declare function cancelAnimationFrame(requestId: AnimationFrameID): void;
                                                          ^^^^^^^^^^^^^^^^ [2]


Error --------------------------------------------------- node_modules/react-beautiful-dnd/src/state/dimension.js:132:10

Cannot return object literal because inexact object literal [1] is incompatible with exact `DroppableDimension` [2].

   node_modules/react-beautiful-dnd/src/state/dimension.js:132:10
                 v
   132|   return {
   133|     ...droppable,
   134|     viewport,
   135|   };
          ^ [1]

References:
   node_modules/react-beautiful-dnd/src/state/dimension.js:85:4
    85| ): DroppableDimension => {
           ^^^^^^^^^^^^^^^^^^ [2]


Error ----------------------------------------------------- node_modules/react-beautiful-dnd/src/state/reducer.js:270:34

Cannot assign object literal to `newDrag` because inexact object literal [1] is incompatible with exact `DragState` [2].

   node_modules/react-beautiful-dnd/src/state/reducer.js:270:34
                                         v
   270|       const newDrag: DragState = {
   271|         ...existing,
   272|         current: {
   273|           ...existing.current,
   274|           hasCompletedFirstBulkPublish: true,
   275|         },
   276|       };
              ^ [1]

References:
   node_modules/react-beautiful-dnd/src/state/reducer.js:270:22
   270|       const newDrag: DragState = {
                             ^^^^^^^^^ [2]


Error ------------------------ node_modules/react-beautiful-dnd/src/view/drag-handle/sensor/create-touch-sensor.js:99:38

Cannot assign `setTimeout(...)` to `longPressTimerId` because `TimeoutID` [1] is incompatible with number [2].

   node_modules/react-beautiful-dnd/src/view/drag-handle/sensor/create-touch-sensor.js:99:38
    99|     const longPressTimerId: number = setTimeout(startDragging, timeForLongPress);
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

References:
   /private/tmp/flow/flowlib_5c03ba9/core.js:752:4
   752| ): TimeoutID;
           ^^^^^^^^^ [1]
   node_modules/react-beautiful-dnd/src/view/drag-handle/sensor/create-touch-sensor.js:99:29
    99|     const longPressTimerId: number = setTimeout(startDragging, timeForLongPress);
                                    ^^^^^^ [2]


Error ----------------------- node_modules/react-beautiful-dnd/src/view/drag-handle/sensor/create-touch-sensor.js:111:18

Cannot call `clearTimeout` with `state.longPressTimerId` bound to `timeoutId` because:
 - null or undefined [1] is incompatible with `TimeoutID` [2].
 - number [3] is incompatible with `TimeoutID` [2].

   node_modules/react-beautiful-dnd/src/view/drag-handle/sensor/create-touch-sensor.js:111:18
   111|     clearTimeout(state.longPressTimerId);
                         ^^^^^^^^^^^^^^^^^^^^^^

References:
   node_modules/react-beautiful-dnd/src/view/drag-handle/sensor/create-touch-sensor.js:18:21
    18|   longPressTimerId: ?number,
                            ^^^^^^^ [1]
   /private/tmp/flow/flowlib_5c03ba9/core.js:747:43
   747| declare function clearTimeout(timeoutId?: TimeoutID): void;
                                                  ^^^^^^^^^ [2]
   node_modules/react-beautiful-dnd/src/view/drag-handle/sensor/create-touch-sensor.js:18:22
    18|   longPressTimerId: ?number,
                             ^^^^^^ [3]


Error --------------------------------------------- node_modules/react-beautiful-dnd/src/view/moveable/moveable.jsx:93:7

Cannot create `Motion` element because:
 - inexact `PlainStyle` [1] is incompatible with exact `Position` [2] in the first argument of property `children`.
 - inexact `$Iterable` [3] is incompatible with exact `React.Element` [4] in the return value of property `children`.
 - inexact `React.Portal` [3] is incompatible with exact `React.Element` [4] in the return value of property `children`.
 - inexact boolean [3] is incompatible with exact `React.Element` [4] in the return value of property `children`.
 - inexact null [3] is incompatible with exact `React.Element` [4] in the return value of property `children`.
 - inexact number [3] is incompatible with exact `React.Element` [4] in the return value of property `children`.
 - inexact string [3] is incompatible with exact `React.Element` [4] in the return value of property `children`.
 - inexact undefined [3] is incompatible with exact `React.Element` [4] in the return value of property `children`.

   node_modules/react-beautiful-dnd/src/view/moveable/moveable.jsx:93:7
             v----------------------------------------------------------------
   93|       <Motion defaultStyle={origin} style={final} onRest={this.onRest}>
   94|         {(current: Position) =>
   95|           this.props.children(
   96|             getStyle(isNotMoving, current.x, current.y)
   97|           )}
   98|       </Motion>
             --------^

References:
   node_modules/react-motion/lib/Types.js.flow:38:33
   38|   children: (interpolatedStyle: PlainStyle) => ReactElement,
                                       ^^^^^^^^^^ [1]
   node_modules/react-beautiful-dnd/src/view/moveable/moveable.jsx:94:20
   94|         {(current: Position) =>
                          ^^^^^^^^ [2]
   node_modules/react-beautiful-dnd/src/view/moveable/moveable-types.js:15:24
   15|   children: (Style) => Node,
                              ^^^^ [3]
   node_modules/react-motion/lib/Types.js.flow:38:48
   38|   children: (interpolatedStyle: PlainStyle) => ReactElement,
                                                      ^^^^^^^^^^^^ [4]



Found 16 errors
```

**Comment by TrySound** [comment]
@SimenB It's because this project uses older version of flow. Did you have ignored paths before?

**Comment by alexreardon** [comment]
We hope to upgrade to the latest flow soon. The setTimeout and
requestAnimationFrame changes are a bit time consuming to work through

**Comment by TrySound** [comment]
@alexreardon I can take care of it.

**Comment by SimenB** [comment]
@TrySound nope, no previous ignores

**Pull request merged**